addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -40,26 +101,55 @@ var CourseJoinWidget = publicWidget.Widget.extend({
             }
         }).popover('show');
     },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
-
     /**
      * @private
-     * @param {MouseEvent} event
      */
-    _onClickJoin: function (event) {
-        event.preventDefault();
-        var channelId = this.channelId || $(event.currentTarget).data('channel-id');
-        this.joinCourse(channelId); 
+    _renderNotPublic: function () {
+        if (this.channelEnroll === 'invite') {
+            this.$('b.o_wslides_js_course_private').removeClass('d-none');        
+            this.$('a.o_wslides_js_course_join_link').addClass('d-none');
+            this.$('b.o_wslides_js_course_public').addClass('d-none');
+        }
+    },
```

Comment:
This could be done in template directly
So not sure the start overrides that calls this method are necessary.